### PR TITLE
Configuration via environment variables

### PR DESCRIPTION
**Pull request by dvaldivia**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,28 +14,32 @@ docker build . -t minio/minsql
...
-To run the project you may specify a `.toml` configuration file, if none is specified *MinSQL* will attempt to look for a `config.toml` file.
+To run the project you need to provide the access details for a `Meta Bucket` to store the shared configuration between multiple `MinSQL` instances, the location and access to it should be configured via environment variables when starting MinSQL .
 
-```
-docker run -v config.toml:config.toml minio/minsql config.toml
-```
-A sample configuration file can be found at `config.toml.template`
+##### Docker:
+````bash
+docker run -e METABUCKET_NAME='minsql-meta' -e METABUCKET_ENDPOINT='http://localhost:9000' -e ACCESS_KEY='minio' -e SECRET_KEY='minio123' minio/minlsql
```
Comment (poornas) [inline README.md]: minio/minlsql -> minio/minsql

File: README.md [hunk 2]
```diff
@@ -14,28 +14,32 @@ docker build . -t minio/minsql
...
+### Environment variables
+
+Environment |  Description |
+--- | --- | 
+METABUCKET_NAME | Name of the meta bucket.
+METABUCKET_ENDPOINT | Endpoint.
+ACCESS_KEY | Meta Bucket Access key
+SECRET_KEY | Meta Bucket Secret key
+PKCS12_CERT | *Optional:* location to a pcks12 certificate.
+PKCS12_PASSWORD | *Optional:* password to unlock the certificate.
```
Comment (donatello) [inline README.md]: As a convention we should prefix all our vars with `MINSQL_` so it does not clash with vars used by other programs in an environment.

File: src/query.rs [hunk 3]
```diff
@@ -869,7 +865,7 @@ fn mk_output_line(
                 outstring.push_str(" ");
             }
         }
-        outstring.push_str("\n");
+        //        outstring.push_str("\n");
```
Comment (donatello) [inline src/query.rs]: We can remove unnecessary code.

File: src/config.rs [hunk 4]
```diff
@@ -140,25 +135,92 @@ impl fmt::Display for ConfigurationError {
...
+                e
+            )));
+        }
     };
-    // try to parse the toml string
-    let mut configuration: Config = match toml::from_str(&contents) {
-        Ok(t) => t,
-        Err(e) => return Err(ConfigurationError::new(&format!("{}", e)[..])),
+
+    let access_key: String = match env::var("METABUCKET_ACCESS_KEY") {
```
Comment (donatello) [inline src/config.rs]: Use a constant for env var names as a good practice: https://doc.rust-lang.org/rust-by-example/custom_types/constants.html

**Review by donatello — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -14,28 +14,32 @@ docker build . -t minio/minsql
...
-A bucket to store shared configuration across multiple `MinSQL` instances is needed, the location and access to it should be configured on the `config.toml`.
+### Environment variables
+
+Environment |  Description |
+--- | --- | 
+MINSQL_METABUCKET_NAME | Name of the meta bucket.
+MINSQL_METABUCKET_ENDPOINT | Endpoint.
+MINSQL_ACCESS_KEY | Meta Bucket Access key
+MINSQL_SECRET_KEY | Meta Bucket Secret key
+MINSQL_PKCS12_CERT | *Optional:* location to a pcks12 certificate.
```
Comment (donatello) [inline README.md]: s/pcks12/pkcs12

**Review by donatello — CHANGES_REQUESTED** [review]

File: src/config.rs [hunk 6]
```diff
@@ -17,17 +17,19 @@
...
 use std::fmt;
-use std::fs;
 
 use log::error;
 use serde_derive::{Deserialize, Serialize};
 
-use crate::constants::DEFAULT_SERVER_ADDRESS;
+use crate::constants::{
+    DEFAULT_SERVER_ADDRESS, METABUCKET_ACCESS_KEY, METABUCKET_ENDPOINT, METABUCKET_NAME,
+    METABUCKET_SECRET_KEY, PKCS12_CERT, PKCS12_PASSWORD,
```
Comment (donatello) [inline src/config.rs]: Environment variable names are being used to configure the server, so they are not needed in other modules. I don't see why they need to be in a separate module and imported here like this. It is simpler to just put the constants in the config module.

**Review by donatello — CHANGES_REQUESTED** [review]

File: src/meta.rs [hunk 7]
```diff
@@ -135,13 +130,21 @@ impl Meta {
                                                     Err(_) => MetaConfigObject::Unknown,
                                                 }
                                             }
```
Comment (poornas) [inline src/meta.rs:132]: there is a panic on line 88 if the metabucket is empty
```
thread 'tokio-runtime-worker-2' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:345:21
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
  9: core::option::Option<T>::unwrap
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libcore/macros.rs:12
  10: minsql::meta::Meta::load_config_from_metabucket::{{closure}}
             at src/meta.rs:88
```

**Review by poornas — DISMISSED** [review]

File: src/meta.rs (hunk 7 above)
Comment (dvaldivia) [reply src/meta.rs:132]: this is fixed on #105

**Review by donatello — APPROVED** [review]

**Pull request merged**